File: QadeerApp.Web/Modules/ServerTypes/Administration/PortalLinkColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PortalLinkRow } from "./PortalLinkRow";

export interface PortalLinkColumns {
    PortalLinkId: Column<PortalLinkRow>;
    DepartmentName: Column<PortalLinkRow>;
    SpecializationName: Column<PortalLinkRow>;
    Title: Column<PortalLinkRow>;
    Url: Column<PortalLinkRow>;
    DisplayOrder: Column<PortalLinkRow>;
    IsActive: Column<PortalLinkRow>;
}

export class PortalLinkColumns extends ColumnsBase<PortalLinkRow> {
    static readonly columnsKey = 'Administration.PortalLink';
    static readonly Fields = fieldsProxy<PortalLinkColumns>();
}

File: QadeerApp.Web/Modules/Administration/PortalLink/PortalLinkColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.PortalLink")]
[BasedOnRow(typeof(PortalLinkRow), CheckNames = true)]
public class PortalLinkColumns
{
    [EditLink, AlignRight]
    public int PortalLinkId { get; set; }

    [Width(160)]
    public string DepartmentName { get; set; }

    [Width(180)]
    public string SpecializationName { get; set; }

    [EditLink, Width(220)]
    public string Title { get; set; }

    [Width(280)]
    public string Url { get; set; }

    [Width(80)]
    public int DisplayOrder { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}
